src/network.rs

```diff
@@ -28,6 +28,7 @@ use crate::{
 /// Wrapper for the pair of a `reqwest::Response` and the `SemaphorePermit` used
 /// to limit concurrent connections, with a test-only variant for mocking.
 enum Response<'a> {
+    #[allow(dead_code)] // It's ok that `SemaphorePermit` is never read.
     Real(reqwest::Response, tokio::sync::SemaphorePermit<'a>),
```

Comment:
```suggestion
    Real(
        reqwest::Response,
        #[allow(unused)] tokio::sync::SemaphorePermit<'a>,
    ),
```

I think I have a mild preference for keeping the scoping as small as possible on the `#[allow]`